openedx/core/djangoapps/auth_exchange/views.py

```diff
@@ -145,12 +158,24 @@ def post(self, request):
         if not hasattr(request.user, 'backend'):
             request.user.backend = self._get_path_of_arbitrary_backend_for_user(request.user)
 
-        if not self._is_grant_password(request.auth):
+        is_grant_password = self._is_grant_password(request)
+
+        if not is_grant_password:
             raise AuthenticationFailed({
                 'error_code': 'non_supported_token',
-                'developer_message': 'Only support DOT type access token with grant type password. '
+                'developer_message': 'Only access tokens with grant type password are supported. '
             })
 
+        if is_jwt_authenticated(request):
+            try:
+                get_asymmetric_only_jwt_decode_handler(request.auth)
+            except jwt.InvalidTokenError as symmetric_jwt_error:
+                error_msg = {
+                    'error_code': 'non_asymmetric_token',
+                    'developer_message': 'Only asymmetric jwt are supported.'
+                }
+                raise AuthenticationFailed(error_msg) from symmetric_jwt_error
```

Comment:
Do you want the test to confirm that you are actually getting this error, and not some other unexpected error?